### ISSUE TITLE
Packaging with Hatchling

### ISSUE THREAD

**Issue by carminoplata** [pypa/packaging-problems#687]
### Problem description

Hi there, 

I tried to follow the tutorial for Python packaging in order to develop using TDD approach my module. I dealt with an issue to package my module using Hatchling build tool. 

I had to add the following lines to my pyproject.toml otherwise my tests weren't able to see the classes and functions declared in the module

`[tool.hatch.build]
include = [
  "src/htmlparser/*.py"
]
exclude = [
  "tests"
]`

I tried to use also the following lines but it didn't work.
`[tool.hatch.build.targets.wheel]
packages = ["src/htmlparser"]`

Is it a Hatch issue? (I'm using python 3.11.4, hatch 1.7.0, hatchling 1.18.0 on Windows 11)

**Comment by henryiii** [comment]
What is the name of your package? If you set `name="htmlparser"`, the above should not be necessary at all. If you need the name of the package and the import name to differ, then I think only the final item is needed:

```toml
[tool.hatch.build.targets.wheel]
packages = ["src/htmlparser"]
```

If you have a link to repo that's helpful. :)

Also, [scientific-python/cookie](https://github.com/scientific-python/cookie) (based on [the Scientific Python Development Guide](https://learn.scientific-python.org/development/guides) can make a new repo for you with a bunch of different backends, including hatching, if you'd like to use that as a starting point.

**Comment by carminoplata** [comment]
Yes, The name of my package is **htmlparser**. Without the include lines and making the editable installation, the tests weren't able to find the module. 
Also with packages field as you said, it didn't work.
Regarding the repo, I added you as Collaborator. Let me know if you have any problem to see the code
